src/main/kotlin/com/exactpro/th2/ws/client/api/impl/WebSocketClient.kt

```diff
@@ -195,7 +195,9 @@ class WebSocketClient(
             } else {
                 logger.info { "Trying to close socket gracefully" }
 
-                handler.runCatching(IHandler::preClose).onFailure {
+                handler.runCatching {
+                    preClose(this@WebSocketClient)
```

Comment:
You won't be able to send data at this moment because the `isRunning` flag is set to `false` and will cause the client to restart. We don't need that when we call the `stop` method. I think you need to call the `preClose` before going into `socket.runCatching {}` block.

By the way, why do you need to send the message before the connection is closed?

Reply:
Good point. 

We have some apis that need to send last requests if possible before disconnect